deploy.sh

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env bash
+set -eo pipefail
+set -x
+
+VAULT_TOKEN="$1"
+GIT_BRANCH="$2"
+DEPLOY_ENV="$3"
+
+set +x
+
+if [[ -z "${DEPLOY_ENV}" ]]; then
+    echo "DEPLOY_ENV argument not supplied; inferring from GIT_BRANCH '${GIT_BRANCH}'."
+
+    #TODO
+    if [[ "${GIT_BRANCH}" == "main" ]]; then
+        DEPLOY_ENV="dev"
+    elif [[ "${GIT_BRANCH}" == "alpha" ]]; then
+        DEPLOY_ENV="alpha"
+    elif [[ "${GIT_BRANCH}" == "perf" ]]; then
+        DEPLOY_ENV="perf"
+    elif [[ "${GIT_BRANCH}" == "staging" ]]; then
+        DEPLOY_ENV="staging"
+    else
+        echo "Git branch '${GIT_BRANCH}' is not configured to automatically deploy to an environment"
+        exit 1
+    fi
+fi
+
+set -u
+
+APP_PATH=/app
+SERVICE_ACCT_KEY_FILE="deploy_account.json"
+GCLOUD_IMAGE="google/cloud-sdk:latest"
+DEPLOY_PROJECT_NAME="broad-dsde-${DEPLOY_ENV}"
+
+if [[ "${DEPLOY_ENV}" =~ ^(dev|alpha|perf|staging)$ ]]; then
+    SOURCE_ENV="${DEPLOY_ENV}"
+else
+    echo "Unknown environment: ${DEPLOY_ENV} - must be one of [dev, alpha, perf, staging]"
+    exit 1
+fi
+
+echo "Deploying branch '${GIT_BRANCH}' from ${SOURCE_ENV} to ${DEPLOY_PROJECT_NAME}"
+set -x
+
+# Get the tier specific credentials for the service account out of Vault
+# Put key into SERVICE_ACCT_KEY_FILE
+docker run \
+    --rm \
+    --env "VAULT_TOKEN=${VAULT_TOKEN}" \
+    broadinstitute/dsde-toolbox \
+    vault read \
+    --format=json "secret/dsde/firecloud/${DEPLOY_ENV}/deltalayer/sa/deltalayer-dev-deployer.json" |
+jq .data > "${SERVICE_ACCT_KEY_FILE}"
+
+docker run \
+    --rm \
+    --entrypoint="/bin/bash" \
+    --volume "$PWD:${APP_PATH}" \
+    --env BASE_URL="https://us-central1-deltalayer-${DEPLOY_ENV}.cloudfunctions.net" \
+    "${GCLOUD_IMAGE}" \
+    -c \
+    "gcloud auth activate-service-account --key-file ${APP_PATH}/${SERVICE_ACCT_KEY_FILE} &&
+      gcloud config set project ${DEPLOY_PROJECT_NAME} &&
+      cd ${APP_PATH} &&
+      gcloud functions deploy delta-layer-listener --trigger-bucket=terra-deltalayer-source-${SOURCE_ENV} \\
+        --entry-point=bio.terra.cloudfiletodatastore.deltalayer.functions.DeltaLayerRawFunction \\
+        --memory 512MB --source=. --runtime java11 --project ${DEPLOY_PROJECT_NAME} \\
+        --service-account deltalayer-${SOURCE_ENV}-streamer@broad-dsde-${SOURCE_ENV}.iam.gserviceaccount.com"
```

Comment:
for my own clarity: this is deploying from source, not from the shadow jar, right? Therefore the changes to downgrade the shadow library above are not used by this? If so, it would be nice (future PR or this one) to skip that step of the build when deploying `DeltaLayerRawFunction`

Reply:
Ultimately, we want to publish the shadow jar to a central bucket located in `dsp-artifact-registry` project. The `gcloud functions deploy` should pull the jar from the central bucket.